src/Raven.Server/Documents/Queries/Graph/RecursionQueryStep.cs

```diff
@@ -109,9 +109,16 @@ public bool IsEmpty()
             return _results.Count == 0;
         }
 
+        public bool CollectIntermediateResults { get; set; }
+
+        public List<Match> IntermediateResults => CollectIntermediateResults ? _results : new List<Match>();
```

Comment:
In fact, why do we need it to be different from `_results` ?

Reply:
collection destination doesn't have _result and we need to collect the requested documents.